homeassistant/components/miraie_ac/climate.py

```diff
@@ -0,0 +1,303 @@
+"""Support for the MirAIe climate."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from py_miraie_ac import (
+    Device as MirAIeDevice,
+    MirAIeAPI,
+    PowerMode as MirAIePowerMode,
+    SwingMode as MirAIeSwingMode,
+)
+
+from homeassistant.components.climate import (
+    FAN_OFF,
+    PRECISION_WHOLE,
+    PRESET_NONE,
+    SWING_BOTH,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+    ClimateEntity,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN
+from .const import (
+    FAN_MODE_MAP_TO_HASS,
+    FAN_MODE_MAP_TO_MIRAIE,
+    HVAC_MODE_MAP_TO_HASS,
+    HVAC_MODE_MAP_TO_MIRAIE,
+    MAX_TEMP,
+    MIN_TEMP,
+    PRESET_MODE_MAP_TO_HASS,
+    PRESET_MODE_MAP_TO_MIRAIE,
+    SUPPORTED_FAN_MODES,
+    SUPPORTED_FEATURES,
+    SUPPORTED_HVAC_MODES,
+    SUPPORTED_PRESET_MODES,
+    SUPPORTED_SWING_MODES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add MirAIe AC devices."""
+    try:
+        api: MirAIeAPI = hass.data[DOMAIN][entry.entry_id]
+        await api.initialize()
+        devices: list[MirAIeDevice] = api.devices
+
+        for device in devices:
+            _LOGGER.debug("Found MirAIe device: %s", device.friendly_name)
+
+        entities = list(map(MirAIeClimateEntity, devices))
+        async_add_entities(entities)
+    except Exception as exp:
```

Comment:
Handling the bare exception is only allowed in the config flow. Please handle only exceptions raised by the API.